internal/relayermetrics/metrics.go

```diff
@@ -38,82 +46,93 @@ func updateMetrics(updateFn func()) {
+// ObserveFinishedTaskExecutionTime records the execution time (ms) of a finished task.
+func ObserveFinishedTaskExecutionTime(
+	tunnelID uint64,
+	destinationChain string,
+	finishedTaskExecutionTime int64,
+) {
 	updateMetrics(func() {
-		metrics.TaskExecutionTime.WithLabelValues(fmt.Sprintf("%d", tunnelID)).Observe(taskExecutionTime)
+		metrics.FinishedTaskExecutionTime.WithLabelValues(fmt.Sprintf("%d", tunnelID), destinationChain).
+			Observe(float64(finishedTaskExecutionTime))
 	})
 }
 
-// IncTunnelsPerDestinationChain increments the total number of tunnels per specific destination chain.
+// IncTunnelsPerDestinationChain increments the count of tunnels per destination chain.
 func IncTunnelsPerDestinationChain(destinationChain string) {
 	updateMetrics(func() {
 		metrics.TunnelsPerDestinationChain.WithLabelValues(destinationChain).Inc()
 	})
 }
 
 // IncActiveTargetContractsCount increases the count of active target contracts.
-func IncActiveTargetContractsCount() {
+func IncActiveTargetContractsCount(destinationChain string) {
 	updateMetrics(func() {
-		metrics.ActiveTargetContractsCount.Inc()
+		metrics.ActiveTargetContractsCount.WithLabelValues(destinationChain).Inc()
 	})
 }
 
 // DecActiveTargetContractsCount decreases the count of active target contracts.
-func DecActiveTargetContractsCount() {
+func DecActiveTargetContractsCount(destinationChain string) {
 	updateMetrics(func() {
-		metrics.ActiveTargetContractsCount.Dec()
+		metrics.ActiveTargetContractsCount.WithLabelValues(destinationChain).Dec()
 	})
 }
 
-// IncTxsCount increments the transactions count metric for a specific tunnel.
-func IncTxsCount(tunnelID uint64) {
+// IncTxsCount increments the transactions count.
+func IncTxsCount(tunnelID uint64, destinationChain string, txStatus string) {
 	updateMetrics(func() {
-		metrics.TxsCount.WithLabelValues(fmt.Sprintf("%d", tunnelID)).Inc()
+		metrics.TxsCount.WithLabelValues(fmt.Sprintf("%d", tunnelID), destinationChain, txStatus).Inc()
 	})
 }
 
-// ObserveTxProcessTime tracks transaction processing time in seconds with millisecond precision.
-func ObserveTxProcessTime(destinationChain string, txProcessTime float64) {
+// ObserveTxProcessTime records the processing time (ms) for each transaction.
+func ObserveTxProcessTime(tunnelID uint64, destinationChain string, txStatus string, txProcessTime int64) {
 	updateMetrics(func() {
-		metrics.TxProcessTime.WithLabelValues(destinationChain).Observe(txProcessTime)
+		metrics.TxProcessTime.WithLabelValues(fmt.Sprintf("%d", tunnelID), destinationChain, txStatus).
+			Observe(float64(txProcessTime))
 	})
 }
 
-// ObserveGasUsed tracks gas used for the each relayed transaction.
-func ObserveGasUsed(tunnelID uint64, gasUsed uint64) {
+// ObserveGasUsed tracks the amount of gas used for each transaction.
+func ObserveGasUsed(tunnelID uint64, destinationChain string, txStatus string, gasUsed decimal.NullDecimal) {
```

Comment:
uint64 should be fine since other chain types might not use this type. standard type should be better.